src/pp_ast.mli

```diff
@@ -0,0 +1,27 @@
+open! Import
+
+module Config : sig
+  type t
+  (** Type for AST pretty-printing config *)
+
+  val make : ?show_attrs:bool -> unit -> t
+  (** Create a custom pretty-printing config.
+
+      Default values are the ones that are used when no configuration is passed
+      to the pretty-printers defined below.
+
+      @param show_attrs
+        controls whether attributes are shown or hidden. It defaults to [false].
+        When set to [true], records such as [expression] that have a [desc]
```

Comment:
I'm a bit confused by this explanation, in particular

> When set to [true], records such as [expression] that have a [desc] field will only be printed if the list of attributes is non-empty

```suggestion
        When set to [true], attribute fields on records such as [expression] that have a [desc]
```

And even if the attributes are empty, it still prints them (i.e. it prints `[]` and not `__attrs`)

Reply:
Yeah this is very poorly phrased indeed. What I was trying to say was:

1. If you don't pass `--show-attr=true`, then records with a `_desc` field are directly printed as the value of said field.
2. If you pass `--show-attr=true`, records with a `_desc` field and an empty list of attributes are still printed as the value of the `_desc` field.
3. If you pass `--show-attr=true`, records with a `_desc` field and a non empty list of attributes are printed as a regular record would.

> And even if the attributes are empty, it still prints them (i.e. it prints [] and not __attrs)

That's true but the behaviour is governed at the record level, i.e. if you look at how `expression`, it's defined using this function https://github.com/ocaml-ppx/ppxlib/pull/517/files#diff-cc3c0a60fe8fd76db7bb4c7d196c81e1a969784fbc96201434a5c3017aa0578aR104 which handles how to print `_desc` field containing records based on the value of `show_attrs` and the presence of `attributes`.

I'd gladly take a better phrasing to describe this behaviour. It might be best to describe it at a higher level of documentation for `ppxlib-pp-ast`, especially since one could be surprised not to see those records "skipped" regardless of the `--show-attr` specific behaviour.

I'll try to do that!